common/ast/src/parser/token.rs

```diff
@@ -0,0 +1,1117 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use logos::Logos;
+use logos::Span;
+
+pub use self::TokenKind::*;
+
+#[allow(non_camel_case_types)]
+#[derive(Logos, Clone, Copy, Debug, PartialEq)]
+pub enum TokenKind {
+    #[error]
```

Comment:
What about reserve a `TokenKind::Error` for this?

Reply:
This is required by `logos` according to its documentation.

Reply:
```rust
...
    #[error]
    Error,

    #[regex(r"[ \t\n\f]+", logos::skip)]
    Whitespace,
...
```

This can work fine. And IMO it's more reasonible to give invalid token a `Error` instead.

Reply:
Addressed